subprojects/GOPATH/src/github.com/microsoft/hivedscheduler/pkg/algorithm/hived_algorithm.go

```diff
@@ -95,7 +111,18 @@ func (h *HivedAlgorithm) UpdateNode(oldNode, newNode *core.Node) {
 }
 
```

Comment:
Check node ready like:

[https://github.com/microsoft/pai/blob/c46df1d917ead84ce5426c7c433af060674201da/src/rest-server/src/models/v2/virtual-cluster/k8s.js#L48](https://github.com/microsoft/pai/blob/c46df1d917ead84ce5426c7c433af060674201da/src/rest-server/src/models/v2/virtual-cluster/k8s.js#L48)


You can track node by yourself (using Add/Update/Del Node), or just pass nodelist insider the scheduler.go to here, then call nodelist.get(node)

Reply:
Why periodically check:
 for {
  s.watchHealthyNodes()
  s.watchBadNodes()
  time.Sleep(watchNodesInterval * time.Second)
 }

For more efficient and clean way is : 
You can track node by yourself (using Algo's Add/Update/Del Node), or just pass nodelister insider the scheduler.go to here, then directly call nodelister.get(node).

Besides, we need to pass the whole node object to algo, because it will also contains which gpu within the heathy node is bad, in future.

---
In reply to: [368357444](https://github.com/microsoft/pai/pull/4153#discussion_r368357444) [](ancestors = 368357444)

Reply:
Also need to check node condition like:

    // check node readiness
    for (let i = node.status.conditions.length - 1; i >= 0; i --) {
      const condition = node.status.conditions[i];
      if (condition.type === 'Ready' && condition.status !== 'Unknown') {
        return true;
      }
    }
    return false;

---
In reply to: [388108886](https://github.com/microsoft/pai/pull/4153#discussion_r388108886) [](ancestors = 388108886,368357444)

Reply:
In future,we can also check one gpu badness by something like ExtractNodeGpuHealthInfo, like ExtractPodBindInfo

---
In reply to: [388133237](https://github.com/microsoft/pai/pull/4153#discussion_r388133237) [](ancestors = 388133237,388108886,368357444)